ansible/group_vars/all.yml

```diff
@@ -40,6 +40,18 @@ insecure_registrys:
 # The port that the Kubernetes apiserver component listens on.
 kube_master_api_port: 443
 
+# Required for HA master configuration
+# Kubernetes master apiserver interface
+#kube_apiserver_interface: eth1
+
+# Required for HA master configuration
+# A floating ip address for multiple kube-apiservers
+# Set to an address that is routeable from all nodes in the cluster
+#kube_apiserver_vip: 172.32.128.100
+
+# The hostname or IP address of the primary apiserver, or VIP for in an HA configuration.
+kube_apiserver_endpoint: "{% if groups['masters']|length > 1 %}{{ kube_apiserver_vip }}{% else %}{{ groups['masters'][0] }}{% endif %}"
```

Comment:
Thanks for the clarification. Even with my new understanding, I still feel that we should not introduce this var. At I high level s/kube_apiserver_endpoint/kube_apiserver_vip

Even if kube-api on master 0 needs to communicate with kube-api, it should go through the lb to hit the api.

Keep in mind that the masters role should have a dep (meta/main.yml) on the lb role when masters is greater than 1.